### DISCUSSION TITLE
Is there any reason not to add F# as a supported language to nvim-treesitter?

### DISCUSSION THREAD

**Discussion by mattgallagher92** [ionide/tree-sitter-fsharp#69] [General]
I found the instructions on getting started with tree-sitter-fsharp in Neovim unclear, and raised #68 to make it easier for people in future.

Rather than having to go through custom installation steps to get F# Treesitter integration in Neovim, it would be easier if F# was added as a supported language to [nvim-tree-sitter](https://github.com/nvim-treesitter/nvim-treesitter/), using tree-sitter-fsharp. That would simplify the installation instructions in this project's README too! Is there any reason not to do that?

**Comment by Nsidorenco** [comment]
The only reason that I have not done it is that I did not want to commit myself to maintain the queries in the nvim tree-sitter repository.

If you are up for the task of adding the parser and queries then I would very much welcome it!

**Reply by mattgallagher92** [reply]
Oh right! I'll admit to some naivety here - I assumed that there was some more automated process than just manually copying the query files into the nvim-treesitter repo 😅 Completely understand not wanting to commit to that 🙂 

I only  started experimenting with Neovim a few days ago, so I'll wait a few weeks to make sure that I'm happy to use it as my daily driver before committing myself!

**Reply by mattgallagher92** [reply]
Some relevant extra context that I've learnt since starting this discussion: I had assumed that query files were build assets, but it seems that they are written by hand and are editor-specific. So, maintaining Neovim integration means manually keeping the query files in sync as the grammar/parser/scanner are updated.

@Nsidorenco please correct me if I'm wrong!

**Reply by Nsidorenco** [reply]
Yep, you hit the nail on the head here!

The queries themselves are entirely custom and to some degree specific to the editor that uses them.
In this case, I have used neovim to make the queries so the query files matches the neovim capture groups.

At this point the work required to maintain official neovim bindings is continually updating the pinned version of this parser in the neovim repo and updating the query files accordingly.
